backend/src/apiserver/resource/resource_manager.go

```diff
@@ -318,18 +318,10 @@ func (r *ResourceManager) CreateRun(apiRun *api.Run) (*model.RunDetail, error) {
 	// Get workflow from either of the two places:
 	// (1) raw pipeline manifest in pipeline_spec
 	// (2) pipeline version in resource_references
-	// And the latter takes priority over the former
-	var workflowSpecManifestBytes []byte
-	err := ConvertPipelineIdToDefaultPipelineVersion(apiRun.PipelineSpec, &apiRun.ResourceReferences, r)
-	if err != nil {
-		return nil, util.Wrap(err, "Failed to find default version to create run with pipeline id.")
-	}
-	workflowSpecManifestBytes, err = r.getWorkflowSpecBytesFromPipelineVersion(apiRun.GetResourceReferences())
+	// And the latter takes priority over the former when the pipeline manifest is from pipeline_spec.pipeline_id
```

Comment:
Shall we add a comment here saying "manifest and pipeline id/version will not exist at the same time, guaranteed by the validation phase"?

Reply:
done